### PR TITLE
Python: Support built-ins in API graphs

### PR DESCRIPTION
**Pull request by tausbn**
Adds support for accessing built-ins via the API graph. These are placed as members of the `builtins` module, mimicking how this works in Python 3.

In Python 2, this module is called `__builtin__` and we silently alias this to `builtins`. In principle this may cause surprise if people try to write, say, `API::moduleImport("__builtin__").getMember("open")`, but ideally they should be using `API::builtin("open")` for this anyway (and with Python 2 being officially deprecated, I see little reason to support this very narrow use-case).

At present, this implementation does not try to account for redefinitions of built-ins (as witnessed by the test with the `SPURIOUS` annotation).

Also, there is some weirdness in one of the tests (the one that has the `MISSING` annotation), but this appears to be unrelated to the present implementation, and more likely due to missing flow into locally-defined functions.

### CODE REVIEW

File: python/ql/src/semmle/python/ApiGraphs.qll [hunk 1]
```diff
@@ -300,11 +303,18 @@ module API {
...
+        ) and
+        // Ignore the following module name, as we alias `__builtin__` to `builtins` elsewhere
+        name != "__builtin__"
         or
-        // When we `import foo.bar.baz` we want to create API graph nodes also for the prefixes
-        // `foo` and `foo.bar`:
-        name = any(ImportExpr e | not e.isRelative()).getAnImportedModuleName()
+        // The `builtins` module should always be implicitly available
+        name = "builtins"
       } or
```
Comment (yoff) [inline python/ql/src/semmle/python/ApiGraphs.qll:318]: I cannot add a suggestion to deleted lines(?!) It is not of great import, but I wonder if it reads simpler like this:
```
MkModuleImport(string name) {
  // Ignore the following module name, as we alias `__builtin__` to `builtins` elsewhere
  name != "__builtin__" and
  (
     // The `builtins` module should always be implicitly available
     name = "builtins"
     or
     imports(_, name)
     or
     // When we `import foo.bar.baz` we want to create API graph nodes also for the prefixes
     // `foo` and `foo.bar`:
     name = any(ImportExpr e | not e.isRelative()).getAnImportedModuleName()
  )
} or
```

**Review by yoff — APPROVED** [review]
LGTM. One possible rewrite suggested. Also, should we gate the aliasing of `__builtin__` on the Python version?

**Comment by tausbn** [comment]
> [...] should we gate the aliasing of `__builtin__` on the Python version?

\*sigh\* I guess we really must, since you _could_ have a module called `__builtin__.py` in Python 3. :man_facepalming:

File: python/ql/src/semmle/python/ApiGraphs.qll (hunk 1 above)
Comment (tausbn) [reply python/ql/src/semmle/python/ApiGraphs.qll]: :+1: Done.

**Comment by tausbn** [comment]
> [...]  (and with Python being officially deprecated, [...]

Just noticed this... I of course mean _Python 2_. :rofl: 
(I edited it to avoid confusion.)

**Review by yoff — APPROVED** [review]
LGTM

**Pull request merged**